### ISSUE TITLE
Variables Created in IF-ELSE Raise redefined-variable-type

### ISSUE THREAD

**Issue by SenorPez** [PyCQA/pylint#997]
### Steps to reproduce
1. I created a quick test script (since it's harder to pull from a full script I'm working on) to distill the problem down:

``` python
class Test():
    def __init__(self, value, use_int=True):
        if use_int:
            self.value = int(value)
        else:
            self.value = float(value)
```
1. Running pylint on this produces:

`R:  6,12: Redefinition of self.value type from int to float (redefined-variable-type)`
### Current behavior

Raises error.
### Expected behavior

Should not raise error if defined in an IF-ELSE block. Seems like this was the intent of #742, but didn't actually happen?
### pylint --version output

```
$ pylint --version
pylint 1.6.1,
astroid 1.4.7
Python 3.4.2 (default, Oct  8 2014, 10:45:20)
[GCC 4.9.1]
```

**Comment by moylop260** [comment]
What about use `self.value = int(value) if use_int else float(value)`?

**Comment by moylop260** [comment]
FYI I can't reproduce it with pylint 2.0.0

**Comment by SenorPez** [comment]
`self.value = int(value) if use_int else float(value)` does not produce the error. However, that precludes the use of other logic in the if/else blocks. In the actual case I saw this error, there's other operations that happen before setting the value.

**Comment by moylop260** [comment]
Could you try reproduce it with pylint 2.0, installing with pip from github?
![screen shot 2016-07-13 at 5 22 11 pm](https://cloud.githubusercontent.com/assets/6644187/16821865/5cdd9432-491e-11e6-966e-250c5cf83bf8.png)

**Comment by PCManticore** [comment]
This is not reproducing on Pylint 2.0, because the check was moved into an extension, due to the huge number of false positives it had. The issue is valid from this point of view and we should check if it still reproduces on 2.0 with the extension enabled.

But for the record, I am not interested in improving this check, I think it's design is bad, currently and it will need a full, more sane, rewrite in order to be useful.

**Comment by PCManticore** [comment]
Closing this. If anyone is willing to volunteer on improving ``redefined-variable-type``, that will be great. Feel free to ignore this error until 2.0 comes out.
